### PR TITLE
feat(eth): Use geth.node for RPC stuff

### PR DESCRIPTION
**Pull request by itsdevbear**
Reuse geths node struct in order to prevent the host chain needing to manage anything rpc related, kinda fire, still a WIP.

Cosmos side affect is that:
- `http://localhost:1317/eth/rpc` -> `http://localhost:8545`
- `ws://localhost:1317/eth/rpc/ws`  -> `ws://localhost:8546` 

But like I'm fine with it. Less stuff for the host chain to have to worry about is bullish.

### CODE REVIEW

**Comment by itsdevbear** [comment]
TODO: Cleanup config and setup stuff.

**Comment by itsdevbear** [comment]
The biggest annoyance on the setup front is that the ChainConfig isn't ready until InitGenesis and that the TxPoolPlugin needs a clientCtx, which forces us to do this weird dance of waiting until shit is ready before sending it down into the keeper.

File: cosmos/precompile/auth/auth.go [hunk 1]
```diff
@@ -32,10 +32,11 @@ import (
 	"pkg.berachain.dev/polaris/cosmos/precompile"
 	"pkg.berachain.dev/polaris/eth/common"
 	ethprecompile "pkg.berachain.dev/polaris/eth/core/precompile"
-	"pkg.berachain.dev/polaris/eth/params"
 	"pkg.berachain.dev/polaris/lib/utils"
 )
 
+const requiredGas = 5000
```
Comment (calbera) [inline cosmos/precompile/auth/auth.go:38]: arbitrary? should probably cost around 1 kvstore read/write?

File: cosmos/runtime/app.go [hunk 2]
```diff
@@ -459,8 +459,7 @@ func (app *PolarisApp) RegisterAPIRoutes(apiSvr *api.Server, apiConfig config.AP
 	if err := server.RegisterSwaggerAPI(apiSvr.ClientCtx, apiSvr.Router, apiConfig.Swagger); err != nil {
 		panic(err)
 	}
-	// Register Ethereum JSON-RPC API as needed by polaris.
-	rpc.RegisterJSONRPCServer(apiSvr.ClientCtx, apiSvr.Router, app.EVMKeeper.GetRPCProvider())
+	app.EVMKeeper.SetClientCtx(apiSvr.ClientCtx)
```
Comment (calbera) [inline cosmos/runtime/app.go:462]: nice

File: eth/provider/config.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/ethereum/go-ethereum/node"
+
+	"pkg.berachain.dev/polaris/eth/rpc"
+)
+
+// DefaultConfig returns the default configuration for the provider.
+func DefaultConfig() *Config {
```
Comment (BrickBera) [inline eth/provider/config.go:35]: 👍

**Pull request merged**